mentions/src/test/kotlin/io/spine/examples/pingh/mentions/GitHubClientSpec.kt

```diff
@@ -0,0 +1,197 @@
+            emitUserLoggedInEventInSessionsContext()
+            val expected = GitHubClient::class.buildBy(gitHubClientId, token)
+            context().assertState(gitHubClientId, expected)
+        }
+    }
+
+    @Nested
+    public inner class `handle 'UpdateMentionsFromGitHub' command, and` {
+
+        private fun sendUpdateMentionsFromGitHubCommand() {
+            val command = UpdateMentionsFromGitHub.newBuilder()
+                .setId(gitHubClientId)
+                .setWhenRequested(currentTime())
+                .vBuild()
+            context().receivesCommand(command)
+        }
+
+        @Test
+        public fun `emits 'MentionsUpdateFromGitHubRequested' event if there is no active update process at this time`() {
+            sendUpdateMentionsFromGitHubCommand()
+            val expected = MentionsUpdateFromGitHubRequested.newBuilder()
+                .setId(gitHubClientId)
+                .vBuild()
+            context().assertEvent(expected)
+        }
+
+        /**
+         * Checks that the value of the `when_started` field is different from the default,
+         * i.e. has been set to some value.
+         *
+         * [MentionsUpdateFromGitHubRequested] command is sent from another thread,
+         * and the main thread checks the state of the entity while performing the update.
+         */
+        @Test
+        public fun `set the start time of the updating operation in 'GitHubClient' entity`() {
+            val otherClientTread = Thread {
+                sendUpdateMentionsFromGitHubCommand()
+            }
+            val gitHubClientWithDefaultWhenStartedField = with(GitHubClient.newBuilder()) {
+                whenStarted = Timestamp.getDefaultInstance()
+                // Building the message partially to include
+                // only the tested fields.
+                buildPartial()
+            }
+            try {
+                otherClientTread.start()
+                context().assertState(gitHubClientId, GitHubClient::class.java)
+                    .ignoringFields(listOf(1, 2))
+                    .isNotEqualTo(gitHubClientWithDefaultWhenStartedField)
+            } finally {
+                otherClientTread.join()
+            }
+        }
+
+        /**
+         * The update request is sent from another thread.
+         *
+         * [MentionsUpdateFromGitHubRequested] command that is sent from another thread
+         * is successfully accepted, after which the command from the main thread is received.
+         * The command from the main thread must be rejected, because
+         * the process started in another thread is not yet complete.
+         */
+        @Test
+        public fun `reject if the update process is already in progress at this time`() {
+            val otherClientTread = Thread {
+                sendUpdateMentionsFromGitHubCommand()
+            }
+            val expectedRejection = MentionsUpdateIsAlreadyInProgress.newBuilder()
```

Comment:
I would move all these `newBuilder()` etc. stuff to private extensions, or even to the test environment.